First draft of the ethics episode (#61)
carpentries-incubator/fair-research-software: episodes/08-research-ethics.md

---

> title: Ethical considerations for research software & data

Author: ```suggestion
title: Ethical considerations for research software
```

teaching: 45
exercises: 15
---

:::::::::::::::::::::::::::::::::::::: questions 

- What ethical considerations are there when using and writing research software (and data)?
- How can our research software (and data) impact the rights and well-being of others?

::::::::::::::::::::::::::::::::::::::::::::::::

::::::::::::::::::::::::::::::::::::: objectives

- Understand some ethical issues around research software development and usage and how they impact others.

::::::::::::::::::::::::::::::::::::::::::::::::

> Issues around research software and data ethics encompass a broad range of concerns related to how 
> software is developed, used, and the data it handles. While these concerns do not strictly fall under
> the FAIR principles, they all contribute to research software quality, reliability and responsibility.
> 

Author: ```suggestion

Research software development and its use (along with data it handles) does not happen in a vacuum - it can pose a number of ethical concerns and challenges for society and the environment, such as contributing to social inequality, compromising privacy and security, increasing environmental impact, and creating social and psychological issues. While these concerns do not strictly fall under
the FAIR principles, they all contribute to research software quality, reliability and responsibility.

```

- Bias mitigation: implement measures to detect and mitigate biases in algorithms and data, 
ensuring that the software treats all users fairly.

TODO: Exercise - perhaps some examples or discussion on algorithm or data bias.

### Environmental impact of research software

- Energy consumption: research software often relies on large-scale data centers for storage and computation, 
or the use of HPC systems and supercomputers for complex simulations and data analysis, which can significantly 
increase energy usage (much of which may come from non-renewable sources).
- Carbon footprint: the electricity used to power data centers and HPC systems contributes to CO2 emissions, 
particularly if the energy comes from fossil fuels.

TODO: Exercise: - e.g. get them to think about optimising algorithms, repeated runs of the untested code, etc.

## Further reading

We recommend the following resources for some additional reading on the topic of this episode:

- ...
- ...

Also check the [full reference set](learners/reference.md#litref) for the course.

:::::::::::::::::::::::::::::::::::::::: keypoints

> - Free and open source research software and data, along with FAIR and ethical considerations, offer viable ways to 

Author: ```suggestion
- Free and open source research software and data, along with FAIR, ethical and environmental considerations, offer viable ways to 
```
